wolfcrypt/src/random.c

```diff
@@ -893,6 +893,11 @@ static int _InitRng(WC_RNG* rng, byte* nonce, word32 nonceSz,
     }
 #endif
 
+#if defined(WOLFSSL_KEEP_RNG_SEED_FD_OPEN) && !defined(USE_WINDOWS_API)
+    if (!rng->seed.seedFdOpen)
+        rng->seed.fd = -1;
```

Comment:
It would be nice if we had an `XBADFD` macro to go with our `XBADFILE` macro, but alas we do not.  If you're feeling inspired, add `XBADFD` in `wc_port.h` and use it here, bearing in mind that it will need to be defined for every target that needs to support `WOLFSSL_KEEP_RNG_SEED_FD_OPEN`.

Reply:
I've only added it for the default target and Zephyr for now, I'm not expecting this macro to be used on embedded systems but XBADFD can easily be expanded if needed.